src/animation_player.rs

```diff
@@ -40,6 +42,16 @@ pub struct HipTransformsEvent {
     pub previous_transform: Vec3,
 }
 
+#[derive(Event)]
+pub struct FrameUpdateEvent {
+    pub time: f32,
+}
+
+#[derive(Resource)]
+pub struct Time {
+    pub time: f32,
+}
+
```

Comment:
be careful of declaring a struct with the name `Time` because bevy has a built-in [`bevy::time::Time`](https://docs.rs/bevy/latest/bevy/time/struct.Time.html) resource as well that keep tracks of the multiple time state e.g. `delta_seconds`, `elapsed_seconds` etc.